server/routes/request.js

```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const protect = require("../middleware/auth");
+const router = express.Router();
+
+const {
+  createRequest,
+  getRequests,
+  updateRequest,
+} = require("../controllers/request");
+
+router.route("/").get(protect, getRequests);
+router.route("/create").post(protect, createRequest);
+router.route("/:id").put(protect, updateRequest);

```

Comment:
`PUT` verb implies a complete replacement of a resource, but your update controller uses `findByIdAndUpdate` method (which by default doesn't necessarily overwrite the entire resource, see the `overwrite` option [here](https://restfulapi.net/resource-naming/)). 

If, by design, you want to encourage the callers do partial updates, we can keep the controller as is and replace the verb with `PATCH`. Otherwise, if you expect a complete replacement of a resource sent with each request, then prefer the `findByIdAndReplace` method.

Finally, [this stackoverflow answer](https://stackoverflow.com/a/19736570) gives good insight on why the distinction might be important for us 👍🏼